### DISCUSSION TITLE
Can lemminx be extended?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse/lemminx-maven#591] [Q&A] [answered]
Sometimes a maven mojo configuration can contain domain specific languages, for example the [bnd-maven-plugin](https://github.com/bndtools/bnd/tree/master/maven-plugins/bnd-maven-plugin) supports embedding of bnd-instructions:

```
<plugin>
    <groupId>biz.aQute.bnd</groupId>
    <artifactId>bnd-maven-plugin</artifactId>
    <configuration>
        <bnd><![CDATA[
            -exportcontents:\
                org.example.api,\
                org.example.types
            -sources: true
        ]]></bnd>
    </configuration>
</plugin>
```

of course lemminx could not understand these, but I wonder if it is possible to add support for such things via a lemminx extension somehow?

So what I want is:

1. Syntax highlighting of the domain specific language
2. Providing validation 
3. Having autocompletions
4. Add code-minings

**Comment by mickaelistria** [comment, marked as answer]
Yes, of course lemminx itself can be extended, especially since lemminx-maven is one of such extensions. See https://github.com/eclipse/lemminx?tab=readme-ov-file#extensions for details.
Such bnd extension could have lemminx-maven as dependency.

**Reply by laeubi** [reply]
Thanks I think that looks like what I was looking for!
